enumerables.rb

```diff
@@ -0,0 +1,141 @@
+module Enumerable
+  def my_each
+    index = 0
+    while index < size
+      if is_a? Array
+        yield self[index]
+      elsif is_a? Range
+        yield to_a[index]
+      end
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    (0...size).each do |i|
+      yield(self[i], i)
+    end
+    self
+  end
+
+  def my_select
+    values = []
+    my_each do |val|
+      values.push(val) if yield(val)
+    end
+    values
+  end
```

Comment:
- [ ] should have working `my_select` identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

Yours is not Returning an enumerator when no block is given.